### PR TITLE
Fix reclique logger

### PR DESCRIPTION
**Pull request by TomScarboroughYMCA**
**Related Issue/Ticket:**

https://github.com/ymcatwincities/openy_gated_content/issues/84

## Steps to test:

- Enable the "Open Y Virtual YMCA Auth ReClique" module for a site using ReClique CORE crm (http://northpennymca-azure-stage.y.org/)
- Configure the module with API user credentials, and use the default Staging endpoint from Reclique
- Attempt member login using any email address - for this repro a legit member email is not required.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [X] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [X] This change does not contain front-end fixes.
- [X] I have flagged this PR "Needs Review" or pinged the VY devs/QA team in Slack

### CODE REVIEW

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php [hunk 1]
```diff
@@ -29,12 +37,15 @@ class RecliqueClientService {
...
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $loggerFactory
+   *   LoggerChannelFactory instance.
    * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
    *   ConfigFactoryInterface instance.
    * @param \GuzzleHttp\Client $client
    *   Guzzle client.
    */
-  public function __construct(ConfigFactoryInterface $configFactory, Client $client) {
+  public function __construct(LoggerChannelFactory $loggerFactory, ConfigFactoryInterface $configFactory, Client $client) {
+    $this->loggerFactory = $loggerFactory->get('openy_gc_auth_reclique');
```
Comment (podarok) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:48]: it's not a factory then, when you are getting a logger via get method
it's a logger

**Review by podarok — CHANGES_REQUESTED** [review]

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php (hunk 1 above)
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:48]:
```suggestion
    $this->logger = $loggerFactory->get('openy_gc_auth_reclique');
```

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php [hunk 2]
```diff
@@ -12,6 +13,13 @@
  */
 class RecliqueClientService {
 
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $loggerFactory;
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:21]:
```suggestion
  protected $logger;
```

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php [hunk 3]
```diff
@@ -72,7 +83,7 @@ public function getUserData($email) {
       }
     }
     catch (\Exception $e) {
-      $this->logger('openy_gated_content')->error($e->getMessage());
+      $this->loggerFactory->error($e->getMessage());
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:86]:
```suggestion
      $this->logger->error($e->getMessage());
```

**Review by anpolimus — CHANGES_REQUESTED** [review]
@TomScarboroughYMCA Andrii is right.
Your variable is a logger itself.
If you approve my suggestion, everything will be fixed.

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php (hunk 2 above)
Comment (TomScarboroughYMCA) [reply modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:21]: Can't argue with that.  :-).  So committed.

**Review by podarok — APPROVED** [review]
Looks great

**Review by anpolimus — APPROVED** [review]

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php (hunk 1 above)
Comment (TomScarboroughYMCA) [reply modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php]: Makes sense - thanks!  I committed Anatoliy's suggested change, so the next build will include this.

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php (hunk 3 above)
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique/src/RecliqueClientService.php:86]:
```suggestion
      $this->logger->error($e->getMessage());
```

**Review by hamrant — APPROVED** [review]

**Review by NightHunterSV — APPROVED** [review]

**Comment by anpolimus** [comment]
Tested at build. Fatal error was fixed

**Pull request merged**